src/Microsoft.DotNet.CodeFormatter.Analyzers.Tests/ProvideExplicitVariableTypeAnalyzerTests.cs

```diff
@@ -300,6 +300,47 @@ void M()
         }
 
         [Fact]
+        public void TestDeclarationWithAnonymousTypeArguments()
+        {
+            const string input = @"
+using System.Linq;
+
+class C1
+{
+    void M()
+    {
+        var names = from i in new [] { 1, 2, 3 }
+                    select new { Number = i, Age = i };
```

Comment:
I see. The type of the variable `names` is `IEnumerable<some_anonymous_type>` -- so `names` is not itself of anonymous type; rather, it is a generic type whose type parameter is anonymous. It took me a minute to understand  how this example tested your new code. Perhaps a comment:

```
// The variable names if of type IEnumerable<some_anonymous_type>. It is not itself of anonymous type,
// but it has a type parameter of anonymous type.